tutorial/tutorial.md

```diff
@@ -0,0 +1,245 @@
+chose the GNU General Public Licence v3.0 or any later version (GPL). More
+than simply choosing a license, you need to put the license in your project
+directory.
+
+You find your license in the [SPDX License List](https://spdx.org/licenses/).
+SPDX is an open standard for communicating license and copyright information.
+Each license is uniquely identified by a shortform [SPDX
+identifier](https://spdx.org/ids). The SPDX identifier for your chosen license
+is GPL-3.0-or-later.
+
+<!-- TODO: Use tool instead -->
+
+You create a `LICENSES` directory in your project root which will contain all
+the licenses that you use in your project. You then download your license
+from the
+[license-list-data](https://github.com/spdx/license-list-data/tree/master/text)
+repository and put it in the `LICENSES` directory.
+
+## 2. Add copyright and license information to each file
+
+Now that you have a license, you need to indicate in the relevant files that
+these files fall under that license. You edit the comment header of
+`src/main.c` as such:
+
+```
+/*
+ * SPDX-Copyright: 2019 Jane Doe <jane@example.com>
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+```
+
+<!-- FIXME: Fix links -->
+
+The `SPDX-Copyright` tag records the publication years and copyright holder
+of the contents of the file. You can read more about [which publication years
+to use]() and [what copyright holders are]() in the FAQ.
+
+The `SPDX-License-Identifier` tag is followed by a [valid SPDX License
+Expression](https://spdx.org/specifications), typically just the SPDX
+Identifier of the license.
+
+<!-- FIXME: Fix links -->
+
+Each file must always contain these two tags in the header. You are allowed
+to [use the tags multiple times if you have multiple copyright holders or
+licenses]().
+
+In the example project, you also edit `Makefile` and `README.md` using this
+header information, but of course with corresponding comment syntax.
+
+### Binary and uncommentable files
+
+You also want to license your image files under GPL-3.0-or-later.
+Unfortunately, images and other binary files do not have comment headers that
+one can easily edit. Other examples include automatically generated files and
+certain data and configuration files for which comments are non-trivial.
+
+There is a simple trick to circumvent this. Create the files `cat.jpg.license`
+and `dog.jpg.license`, each containing the same information about license and
+copyright holder as above.
+
+<!-- TODO: Add link to DEP-5 possibility. -->
+
+### Change licensing information
+
+You discover that the photo of the cat was not licensed under the GPL at all,
+but under Creative Commons Attribution 4.0 International.
+
+The SPDX Identifier of this license is CC-BY-4.0.  You create
+the file `LICENSES/CC-BY-4.0.txt`, following the same steps you used for
+GPL-3.0-or-later.
+
+You then edit `cat.jpg.license` to say:
+
+```
+SPDX-Copyright: 2017 Joe Somebody <joe@example.com>
```

Comment:
Perhaps we should also include the fact that the reader found out that the copyright in the image is held by Peter Janzen

Or perhaps we need better example images.